### PR TITLE
initialization ConnectionTimeouts

### PR DESCRIPTION
**Pull request by CheSema**
fix initialization `http_keep_alive_timeout` in ConnectionTimeouts

also fixes wrond initialization `Poco::Timespan`
- `http_auth_params.timeouts = ConnectionTimeouts{connection_timeout_ms, receive_timeout_ms, send_timeout_ms};`
- `Poco::Timespan(config.getInt("handshake_timeout_ms", DBMS_DEFAULT_RECEIVE_TIMEOUT_SEC * 1000), 0));`

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 34f4ff3db4627bd285ed243ba34ca5f76d9226ed with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59000/34f4ff3db4627bd285ed243ba34ca5f76d9226ed/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59000/34f4ff3db4627bd285ed243ba34ca5f76d9226ed/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/IO/ConnectionTimeouts.h [hunk 1]
```diff
@@ -72,6 +58,57 @@ struct ConnectionTimeouts
...
+
+    return ConnectionTimeouts(*this)
+APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS(SATURATE_MEMBER);
+
+#undef SATURETE_MEMBER
+}
+
+#undef APPLY_FOR_ALL_CONNECTION_TIMEOUT_MEMBERS
+
+inline ConnectionTimeouts & ConnectionTimeouts::withAnyConnectionTimeout(size_t seconds)
```
Comment (antonio2368) [inline src/IO/ConnectionTimeouts.h:102]: I'm having some concerns that people will automatically use `withConnectionTimeout` by default and not set `secure_connection_timeout` by accident
the only time you want to just set `connection_timeout` explicitly is if you want to set the `secure_connection_timeout` so maybe have `withConnectionTimeout` set both timeouts? and you can just override secure timeout with `withSslConnectionTimeout`.

File: src/Disks/IO/ReadBufferFromWebServer.cpp [hunk 2]
```diff
@@ -56,15 +56,16 @@ std::unique_ptr<ReadBuffer> ReadBufferFromWebServer::initialize()
     const auto & settings = context->getSettingsRef();
     const auto & server_settings = context->getServerSettings();
 
+    auto connection_timeouts = ConnectionTimeouts::getHTTPTimeouts(settings, server_settings.keep_alive_timeout);
+    connection_timeouts.withConnectionTimeout(std::max<Poco::Timespan>(settings.http_connection_timeout, Poco::Timespan(20, 0)));
+    connection_timeouts.withSslConnectionTimeout(std::max<Poco::Timespan>(settings.http_connection_timeout, Poco::Timespan(20, 0)));
+    connection_timeouts.withReceiveTimeout(std::max<Poco::Timespan>(settings.http_receive_timeout, Poco::Timespan(20, 0)));
```
Comment (antonio2368) [inline src/Disks/IO/ReadBufferFromWebServer.cpp:62]: why not chain it?

Comment (CheSema) [reply src/Disks/IO/ReadBufferFromWebServer.cpp:62]: don't know the answer)

That code looks suspicious for me.
We take settings but we do not trust it, we make it sure that it is not too low.
It it ether the settings is wrong and other one has to be used, or its default is wrong.

Comment (CheSema) [reply src/Disks/IO/ReadBufferFromWebServer.cpp]: So I don't want to chain them. Just leave it as it -- suspicious.

**Comment by nikitamikhaylov** [comment]
I think the changelog category is definitely not "Not for changelog"

**Comment by CheSema** [comment]
> I think the changelog category is definitely not "Not for changelog"

for `bug` there should be either visible to the user error or broken test. Which we should have have, but we dont. 
I will write a good test for keep alive in a scope of this https://github.com/ClickHouse/ClickHouse/pull/58845

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**